gem/context_processors.py

```diff
@@ -33,12 +33,39 @@ def detect_freebasics(request):
 
 
 def compress_settings(request):
+    REGISTRATION_URL = settings.REGISTRATION_URL
+    EDIT_PROFILE_URL = settings.EDIT_PROFILE_URL
+    VIEW_PROFILE_URL = settings.VIEW_PROFILE_URL
+
+    if settings.USE_OIDC_AUTHENTICATION:
+        site = request.site
+        if not hasattr(site, "oidcsettings"):
+            raise RuntimeError(
+                "Site {} has no settings configured.".format(site))
+
+        oidc_settings = site.oidcsettings
+        if oidc_settings:
+            REGISTRATION_URL = (
+                "%s/registration/?theme=%s&hide=end-user&redirect_uri=%s" % (
+                    settings.OIDC_OP, settings.THEME,
+                    oidc_settings.wagtail_redirect_url))
```

Comment:
should registration not be redirecting to the home page?

Reply:
@sewagodimo  the wagtail redirect url can be set to the homepage in the OIDC settings